### PR TITLE
Add navbar mobile version

### PR DESCRIPTION
**Pull request by ghost**
Description :

This Pull Request adds a mobile version of the navigation bar to improve the user experience on narrow screen devices. Here is an overview of the changes made:

    NavbarMobile Component: A new React NavbarMobile component has been created to specifically manage the layout of the navigation bar on small screens.

    Integration into the Header component: The NavbarMobile component has been integrated into the existing Header component. It is now activated through a menu button, allowing users to easily switch between the standard navigation bar and the mobile version.

    State management: To optimize performance and improve usability, the display of the mobile navigation bar is now conditional, based on the state of the menu button.

    Styles and Interaction: Styles have been adjusted to ensure an aesthetic presentation and smooth user experience when using the mobile version of the navigation bar.

Screenshot:
Screenshot_12

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
 import Banner from "@/components/Banner";
 import Header from "@/components/Header";
+import Image from "next/image";
```
Comment (omjogani) [inline app/page.tsx]: Remove this unused Import!

File: components/Banner.tsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+"use client";
```
Comment (omjogani) [inline components/Banner.tsx:1]: While you're not using any client side rendering functionality, please remove `"use client"`.

File: components/Header.tsx [hunk 3]
```diff
@@ -6,46 +6,60 @@ import Link from "next/link";
 import SearchBar from "./SearchBar";
 import { IconMenu } from "@tabler/icons-react";
 import Headroom from "react-headroom";
+import NavbarMobile from "./NavbarMobile";
+import { useState } from "react";
```
Comment (omjogani) [inline components/Header.tsx:10]: If you're using `useState` you might need to add `"use client"` otherwise client side rendering is not going to work...

**Review by omjogani — CHANGES_REQUESTED** [review]
Please have a look at comments! I'm fine with other stuff!

File: components/Banner.tsx (hunk 2 above)
Comment (ghost) [reply components/Banner.tsx:1]: use client it's necessary otherwise I have a framer motion error

File: components/Header.tsx (hunk 3 above)
Comment (ghost) [reply components/Header.tsx:10]: I use it use client in header.tsx

File: components/Banner.tsx (hunk 2 above)
Comment (omjogani) [reply components/Banner.tsx:1]: Alright, It's required in internal implementation of framer motion.

**Review by omjogani — APPROVED** [review]
Done!

**Pull request merged**